sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -293,11 +296,24 @@ abstract class UnixTimeExprMeta[A <: BinaryExpression with TimeZoneAwareExpressi
     // Date and Timestamp work too
     if (expr.right.dataType == StringType) {
       try {
-        val rightLit = extractStringLit(expr.right)
-        if (rightLit.isDefined) {
-          strfFormat = DateUtils.toStrf(rightLit.get)
-        } else {
-          willNotWorkOnGpu("format has to be a string literal")
+        extractStringLit(expr.right) match {
+          case Some(rightLit) =>
+            if (GpuOverrides.getTimeParserPolicy == LegacyTimeParserPolicy) {
+              willNotWorkOnGpu("legacyTimeParserPolicy is LEGACY")
+            } else {
+              val gpuSupportedFormats = Seq(
+                "yyyy-MM-dd",
+                "yyyy-MM-dd HH:mm:ss"
```

Comment:
Is this a regression in some way?  Did we only support these before?  If not then it would be good to find a way to expand this to something much closer to what we did support before.

Reply:
This is a regression in terms of what will run on GPU but is also an improvement because we are now compatible with Spark for these formats instead of returning bad data for inputs that don't match. I can certainly expand the list of supported formats and maybe make this more general so that we support any fixed-width format. The tricky one is supporting timestamp with millis because cuDF is much stricter than Spark, which will accept partial millis. I will spend more time looking at options here.

Reply:
I am fine with doing that in a follow-on issue.  We are fixing a bug after all.  I mostly wanted to be sure that I understood it correctly and that we don't drop it.

Reply:
I have now changed the behavior so that the expression runs on GPU if a compatible format is being used, or if incompatibleOps is true, so this is no longer a regression.